geode-core/src/main/java/org/apache/geode/management/internal/beans/DataQueryEngine.java

```diff
@@ -0,0 +1,295 @@
+        } else {
+          Object[] functionArgsList = (Object[]) functionArgs;
+          boolean showMember = (Boolean) functionArgsList[DISPLAY_MEMBERWISE];
+          if (showMember) {// Added to show a single member similar to multiple
+            // member.
+            // Note , if no member is selected this is the code path executed. A
+            // random associated member is chosen.
+            List<String> decompressedList = new ArrayList<>();
+            decompressedList.add(BeanUtilFuncs.decompress(result));
+            return wrapResult(decompressedList.toString());
+          }
+          return BeanUtilFuncs.decompress(result);
+        }
+
+      } else { // More than 1 Member
+        ResultCollector coll = FunctionService.onMembers(members).setArguments(functionArgs)
+            .execute(ManagementConstants.QUERY_DATA_FUNCTION);
+
+        List list = (List) coll.getResult();
+        Object object = list.get(0);
+        if (object instanceof Throwable) {
+          throw (Throwable) object;
+        }
+
+        Iterator<byte[]> it = list.iterator();
+        List<String> decompressedList = new ArrayList<>();
+
+        while (it.hasNext()) {
+          String decompressedStr;
+          decompressedStr = BeanUtilFuncs.decompress(it.next());
+          decompressedList.add(decompressedStr);
+        }
+
+        if (zipResult) {
+          return BeanUtilFuncs.compress(wrapResult(decompressedList.toString()));
+        } else {
+          return wrapResult(decompressedList.toString());
+        }
+
+      }
+    } catch (FunctionException fe) {
+      throw new Exception(
+          String.format("Query could not be executed due to : %s", fe.getMessage()));
+    } catch (VirtualMachineError e) {
+      SystemFailure.initiateFailure(e);
+      throw e;
+    } catch (Throwable e) {
+      SystemFailure.checkFailure();
+      throw new Exception(String.format("Query could not be executed due to : %s", e.getMessage()));
+    }
+  }
+
+  private static String wrapResult(final String str) {
+    StringWriter w = new StringWriter();
+    synchronized (w.getBuffer()) {
+      w.write("{\"result\":");
+      w.write(str);
+      w.write("}");
+      return w.toString();
+    }
+  }
+
+  private Set<String> compileQuery(final String query)
+      throws QueryInvalidException {
+    QCompiler compiler = new QCompiler();
+    Set<String> regionsInQuery;
+    try {
+      CompiledValue compiledQuery = compiler.compileQuery(query);
+      Set<String> regions = new HashSet<>();
+      compiledQuery.getRegionsInQuery(regions, null);
+      regionsInQuery = Collections.unmodifiableSet(regions);
+      return regionsInQuery;
+    } catch (QueryInvalidException qe) {
+      logger.error("{} Failed, Error {}", query, qe.getMessage(), qe);
+      throw qe;
+    }
+  }
+
+  static class JsonisedErrorMessage {
+
+    private static String message = "message";
+
+    private GfJsonObject gFJsonObject = new GfJsonObject();
```

Comment:
While you're refactoring this, please could you switch to Jackson here.

Reply:
this is to prepare for that effort. In this code change, I just want to separate the queryData api out of the QueryDataFunction for better understanding of the code structure.

Reply:
No problem.